cmake/Testing.cmake

```diff
@@ -11,8 +11,9 @@ IF(NOT DEFINED ENV{BUILD_ARCHITECTURE} AND EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/te
     target_link_libraries(${UNIT_TEST} PUBLIC 
         ${PROJECT_NAME}
         ${TEST_LINK_LIBRARIES}
-        gtest)
+        GTest::gtest_main)
     add_test(NAME ${UNIT_TEST} COMMAND ${UNIT_TEST})
+    gtest_discover_tests(${UNIT_TEST})
```

Comment:
I think we need to `include(GoogleTest)` before calling this function. That function is defined in the GoogleTest module.

Reply:
It works without including GoogleTest. Is that already included by find_package or something. Is best practice to explicitly include it for readability's sake or something.

Reply:
It probably gets included somewhere further up in the build hierarchy, but I think we should explicitly include it here to avoid relying on transitive includes. This is a best practice.